src/xrScriptEngine/script_engine.cpp

```diff
@@ -754,8 +753,20 @@ void CScriptEngine::lua_error(lua_State* L)
 
 int CScriptEngine::lua_pcall_failed(lua_State* L)
 {
-    print_output(L, "", LUA_ERRRUN);
+#if (!defined(DEBUG) && !XRAY_EXCEPTIONS)
+    print_output(L, "", 0); // Force game to not break in raii_guard() for this type of errors
+#else
+    print_output(L, "", LUA_ERRRUN); // Default behavior
+#endif
```

Comment:
Did i say somewhere that lua_pcall_failed  didn't crash the game? There is ::Fatal() right at the end of lua_pcall_failed body.